code/processes/cmedecoder.q

```diff
@@ -8,16 +8,17 @@
 / process one message from log
 msg:{
   / generate dictionary from message, with correct tags & properly typed values
-  msg:(!/) flip {
+  msg:(!/) {
     / get field name & type from tag number
-    d:exec from .fix.fields where number=x[0];
+    d:.fix.fields each x[;0];
     / check if field has enumerations
-    enum:exec from .fix.enums where name=d[`name];
-    val:$[null enum[`name];x[1];enum[`values]@enum[`enums]?x[1]];
+    enum:.fix.enums each d[`name];
+    val:{$[null x[`name1];y[1];x[`values]@x[`enums]?y[1]]}'[enum;x];
```

Comment:
Think I see the purpose of name1 here, but it seems messy - perhaps a nicer way to do it:

``update name:d`name from .fix.enums each d`name``

for example:

```
q.cme))first update name:d`name from .fix.enums each d`name
enums | ("";"";"";"")
values| ("";"";"";"")
name1 | `
name  | `ApplVerID
```

rather than having an extra field in the table